### PR TITLE
[SPARK-36876][SQL] Support Dynamic Partition pruning for HiveTableScanExec

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Current code just support dynamic partition pruning for DSV1 and DSV2, here we support HiveTableScan

### Why are the changes needed?
Optimize Hive Table Scan dynamic partition pruning 

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
Added UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan

File: sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala [hunk 1]
```diff
@@ -443,115 +447,6 @@ abstract class DynamicPartitionPruningSuiteBase
     }
   }
 
-  /**
-   * The filtering policy has a fallback when the stats are unavailable
-   */
-  test("filtering ratio policy fallback") {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala:449]: this test doesn't work for hive table?

File: sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala [hunk 2]
```diff
@@ -49,7 +49,7 @@ abstract class DynamicPartitionPruningSuiteBase
   protected def initState(): Unit = {}
   protected def runAnalyzeColumnCommands: Boolean = true
 
-  override def beforeAll(): Unit = {
+  override protected def beforeAll(): Unit = {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala:52]: why do we need to change the modifier?

File: sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala (hunk 1 above)
Comment (AngersZhuuuu) [reply sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala:449]: > this test doesn't work for hive table?

Seems hive table stats is not same.

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan

**Review by cloud-fan — APPROVED** [review]

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan

**Review by cloud-fan — APPROVED** [review]

**Pull request closed without merging**